### PR TITLE
[feature] support sysconfig, default, and rc.conf.d

### LINKED ISSUE #17: manage pid and log files
the role does not have the following variables, `kibana_pid_file`, `kibana_pid_dir`, `kibana_log_dir`, and `kibana_log_file`.

similar to https://github.com/reallyenglish/ansible-role-fluentd/issues/38

### PR DESCRIPTION
**Pull request by trombik**
was initially trying to fix #17. however, OpenBSD does not support PID file anymore for good reasons. when you need to know PID of a process, `cat /var/run/foo.pid` is not a good idea because the pid in the file might not what you are looking for.

instead, support `sysconfig`, `default`, and `rc.conf.d`.

### CODE REVIEW

**Comment by trombik** [comment]
@supachots pelase review the PR after reading the proposal. https://github.com/reallyenglish/ansible-role-example/tree/procedure_to_review/docs/Procedures/Reviewing_Pull_Request

File: tests/serverspec/default_spec.rb [hunk 1]
```diff
@@ -26,6 +41,79 @@
...
+  end
+when "freebsd"
+  describe file("/etc/rc.conf.d/kibana") do
+    it { should exist }
+    it { should be_file }
+    it { should be_owned_by default_user }
+    it { should be_grouped_into default_group }
+    it { should be_mode 644 }
+    its(:content) { should match(/^kibana_user="www"$/) }
+    its(:content) { should match(/^kibana_group="www"$/) }
```
Comment (mitsururike) [inline tests/serverspec/default_spec.rb]: s/www/#{kibana_user_group}/

File: tests/serverspec/default_spec.rb [hunk 2]
```diff
@@ -26,6 +41,79 @@
...
+    its(:content) { should match(/^KILL_ON_STOP_TIMEOUT="0"$/) }
+  end
+when "freebsd"
+  describe file("/etc/rc.conf.d/kibana") do
+    it { should exist }
+    it { should be_file }
+    it { should be_owned_by default_user }
+    it { should be_grouped_into default_group }
+    it { should be_mode 644 }
+    its(:content) { should match(/^kibana_user="www"$/) }
```
Comment (mitsururike) [inline tests/serverspec/default_spec.rb]: s/www/#{kibana_user_name}/

File: tasks/configure-Debian.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+
+- name: Create /etc/default/kibana
+  template:
+    src: Ubuntu.default.j2
+    dest: /etc/default/kibana
+    # any files under /etc/default should be mode 644, but the package installs
+    # the file as mode 0664
```
Comment (mitsururike) [inline tasks/configure-Debian.yml]: is it a bug or is there any reason that the package sets 0664?
bug number should be written if it has been reported as a bug.

Comment (trombik) [inline tasks/configure-Debian.yml]: i simply do not know why the mode is 664. not an expert in Debian/Ubuntu in any way, but here is my understanding: files under `/etc/default` are supposed to be sourced by init scripts, or systemd unit files. in rare cases, such as when the content must be protected from non-previledged users, different mode should be used. but 664? no, it does not make sense in this context. if a member of `kibana` needs write access to files under `/etc/default`, that should be a bug.

no, i do not know if the issue has been reported as a bug.

File: tests/serverspec/default_spec.rb [hunk 4]
```diff
@@ -107,8 +107,8 @@
     it { should be_owned_by default_user }
     it { should be_grouped_into default_group }
     it { should be_mode 644 }
-    its(:content) { should match(/^kibana_user="www"$/) }
-    its(:content) { should match(/^kibana_group="www"$/) }
+    its(:content) { should match(/^kibana_user="#{kibana_user_name}"$/) }
+    its(:content) { should match(/^kibana_group="#{kibana_user_name}"$/) }
```
Comment (mitsururike) [inline tests/serverspec/default_spec.rb]: s/name/group/

File: tasks/configure-Debian.yml (hunk 3 above)
Comment (mitsururike) [inline tasks/configure-Debian.yml]: * the explanation above should be added as a comment.
* the issue should be reported to upstream if not yet.

Comment (trombik) [inline tasks/configure-Debian.yml]: will report when time permits.

**Pull request merged**